Lib/inspect.py

```diff
@@ -978,10 +984,18 @@ def getmodule(object, _filename=None):
     # Try the filename to modulename cache
     if _filename is not None and _filename in modulesbyfile:
         return sys.modules.get(modulesbyfile[_filename])
+    # Compute hash to track moduleless objects
+    code = _getcode(object)
+    hashcode = hash(code) if code else None
+    if hashcode and hashcode in _moduleless:
+        return None
     # Try the cache again with the absolute file name
     try:
         file = getabsfile(object, _filename)
     except (TypeError, FileNotFoundError):
+        _moduleless.add(hashcode)
+        return None
+    if hashcode and hashcode in _moduleless:
         return None
```

Comment:
Can this condition ever be True? If is already checked on line 990, and the try- except will return None if a hash is added.

Reply:
I believe so; `getabsfile` can recursively call `getmodule`, and that call can add to `_moduleless` without raising.

Reply:
True. The line could still be removed, as the case is rare and the code will still work, but the extra check is cheap so lets keep it.